src/components/Tabs/Tabs.tsx

```diff
@@ -16,92 +16,102 @@ export interface TabsProps extends BoxProps {
-    as = 'nav',
-    borderWidth = '0 0 xs 0',
-    borderColor = 'grey-100',
-    children,
-    isCentered = false,
-    isFullWidth = false,
-    onChange,
-    overflow = 'auto',
-    size = 'md',
-    value,
-    ...restProps
-  },
-  ref,
-) => {
-  const decoratedChildren = React.Children.map(children, (child, index) => {
-    if (React.isValidElement(child)) {
-      /**
-       * Merging any existing onClick handlers with our onChange handler.
-       */
-      const onClickHandler = (event: React.MouseEvent<HTMLLIElement>) => {
-        if (child.props.onClick) {
-          child.props.onClick(event);
-        }
+const TabsBaseComponent: React.FC<TabsProps> = React.forwardRef(
+  (
+    {
+      as = 'nav',
+      borderWidth = '0 0 xs 0',
+      borderColor = 'grey-100',
+      children,
+      isCentered = false,
+      isFullWidth = false,
+      onChange,
+      overflow = 'auto',
+      size = 'md',
+      value,
+      ...restProps
+    },
+    ref,
+  ) => {
+    const activeTabRef = React.useRef() as React.RefObject<HTMLDivElement>;
+
+    const decoratedChildren = React.Children.map(children, (child, index) => {
+      if (React.isValidElement(child)) {
+        /**
+         * Merging any existing onClick handlers with our onChange handler.
+         */
+        const onClickHandler = (event: React.MouseEvent<HTMLLIElement>) => {
+          if (child.props.onClick) {
+            child.props.onClick(event);
+          }
 
-        if (!child.props.isDisabled && onChange) {
-          onChange(event, index);
-        }
-      };
+          if (!child.props.isDisabled && onChange) {
+            onChange(event, index);
+          }
+        };
 
-      /**
-       * Merge our custom styling with existing className in child
-       */
-      const classes = classNames(
-        child.props.className,
-        styles['tab-item'],
-        { [styles.disabled]: child.props.isDisabled },
-        { [styles['tab-item--selected']]: value === index },
-      );
+        /**
+         * Merge our custom styling with existing className in child
+         */
+        const classes = classNames(
+          child.props.className,
+          styles['tab-item'],
+          { [styles.disabled]: child.props.isDisabled },
+          { [styles['tab-item--selected']]: value === index },
+        );
 
-      return React.cloneElement(
-        child,
-        {
+        return React.cloneElement(child, {
           ...child.props,
           className: classes,
           onClick: onClickHandler,
           fontSize: size,
           padding: size,
           key: child.key,
-          style: { ...child.props.style, ...isFullWidth && { flex: 1 } },
+          style: { ...child.props.style, ...(isFullWidth && { flex: 1 }) },
           'aria-posinset': index + 1,
           'aria-setsize': React.Children.count(children),
           'aria-selected': value === index,
-        },
-      );
-    }
+          ref: value === index ? activeTabRef : null,
```

Comment:
@nathanyoung -- one minor thing here. When we take over a prop in a component (in this case `TabItem`) it's a good idea to merge with any ref the user may have passed in as well for other purposes. Since the TabItem takes in a ref, it is expected that it will work, and in this case it will fail when the tab is active since we hijack it for the scroll behavior. There is a utility function called `mergeRefs` specifically for this purpose. See the `TabsSlider`, `FileUpload`, and `Popover` for examples of usage.